mne/channels/layout.py

```diff
@@ -924,3 +928,69 @@ def apply_montage(info, montage):
                          'names.')
+
+
+def read_montage_polhemus(fname, point_names, kind='Polhemus digitized montage',
+                          transform=True):
+    """Read a Montage from points exported by Polhemus FastScan
+
+    Parameters
+    ----------
+    fname : str
+        Filename of the Polhemus Points file. The file should be exported as
+        text file.
+    point_names : list of str | str
+        A list with the names of all points, in the same order as the points in
+        the file. Alternatively path to a plain text file with one point name
+        per line. Points named with the empty string ('') are ignored.
+    kind : str
+        Name for the Montage.
+    transform : bool
+        Transform the points to MNE's coordinate system. This requires the
+        'Nasion', 'LPA' and 'RPA' points to be specified. When False, the
+        points are still transformed to the RAS coordinate system but are not
+        aligned using Nasion, LPA and RPA.
+
+    Returns
+    -------
+    montage : Montage
+        Montage created from the points.
+    """
+    points = _read_dig_points(fname)
+    if isinstance(point_names, string_types):
+        with open(point_names) as fid:
+            point_names_read = [name.strip() for name in fid.readlines()]
+
+        if len(points) != len(point_names_read):
+            raise ValueError("The file %s contains %i points, but %i names "
+                             "are specified in %s."
+                             % (fname, len(points), len(point_names_read),
+                                point_names))
+        point_names = point_names_read
+    elif len(points) != len(point_names):
+        raise ValueError("The file %s contains %i points, but %i names were "
+                         "specified." % (fname, len(points), len(point_names)))
+    points = apply_trans(als_ras_trans_mm, points)
+
+    if transform:
+        names_lower = [name.lower() for name in point_names]
+
+        # check that all needed points are present
+        missing = [name for name in ('nasion', 'lpa', 'rpa')
+                   if name not in names_lower]
+        if missing:
+            raise ValueError("The points %s are missing, but are needed "
+                             "to transform the points to the MNE coordinate "
+                             "system. Either add the points, or read the "
+                             "montage with transform=False." % missing)
+
+        nasion = points[names_lower.index('nasion')]
+        lpa = points[names_lower.index('lpa')]
+        rpa = points[names_lower.index('rpa')]
+        neuromag_trans = get_ras_to_neuromag_trans(nasion, lpa, rpa)
+        points = apply_trans(neuromag_trans, points)
+
+    # drop points
+    index = np.array([bool(name) for name in point_names])
+    point_names = [name for name in point_names if name]
+    return Montage(points[index], point_names, kind, np.arange(len(points)))
```

Comment:
shouldn't the selection be np.where(points[index])[0] instead of np.arange(len(points))?